src/rrd_parsetime.c

```diff
@@ -832,16 +832,28 @@ static char *day(
 
 static mutex_t parsetime_mutex = MUTEX_INITIALIZER;
 
+static char     *rrd_parsetime_nomt(
+    const char *tspec,
+    rrd_time_value_t * ptv);
+    
 char     *rrd_parsetime(
     const char *tspec,
     rrd_time_value_t * ptv)
 {
-    time_t    now = time(NULL);
-    int       hr = 0;
-
     /* yes this code is non re-entrant ... so lets make sure we do not run
        in twice */
     mutex_lock(&parsetime_mutex);
+    char *result = rrd_parsetime_nomt(tspec, ptv);
+    mutex_unlock(&parsetime_mutex);
```

Comment:
Oh yes, I overlooked this.
Sorry this is my first pull request. How is the workflow now?
And I see, that I also have to think about the EnsureFreeMem at the end.

2015-10-23 15:29 GMT+02:00 Tobias Oetiker <notifications@github.com>:

> In src/rrd_parsetime.c
> <https://github.com/oetiker/rrdtool-1.x/pull/678#discussion_r42864280>:
>
> >      /* yes this code is non re-entrant ... so lets make sure we do not run
> >         in twice */
> >      mutex_lock(&parsetime_mutex);
> > +    char *result = rrd_parsetime_nomt(tspec, ptv);
> > +    mutex_unlock(&parsetime_mutex);
>
> shouldn't you then get rid of the original unlock line ? (1009 after this
> patch)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/oetiker/rrdtool-1.x/pull/678/files#r42864280>.
>